### DISCUSSION TITLE
Is there a cleaner way to write queries?

### DISCUSSION THREAD

**Discussion by ggukggukee** [prisma/prisma#15951] [Q&A]
Hello!

When I write queries for nodejs routing I don't allow users to see more information than they supposed to. 
Therefore I write queries with `select` not `include`.

For example: 
(this is a short example but I have queries that have multiple nested selects and way longer than this)
```js
router.get('/student/:id', authorize, isStudent, async (req, res) => {
  const id = req.params.id;

  try {
    const lessons = await prisma.lesson.findMany({
      where: {
        isDeleted: false,
        userId: parseInt(id),
      },
      select: {
        id: true,
        userId: true,
        groupId: true,
        group: {
          select: {
            id: true,
            number: true
          },
        },
       },
    });

   return res.json(lessons);
})
```

But this method requires too many lines and don't look good. Is there any way to shorten queries like that? 
Any workaround is acceptable. Thank you.

**Comment by austincrim** [comment]
Hey @ggukggukee 👋,

You may be interested in this [outstanding feature request](https://github.com/prisma/prisma/issues/5042). Feel free to add a 👍.

Otherwise, the way you've shown above is the best way to do it for now. If you really don't like this method, you could write use [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw), but that may not be any shorter.

**Reply by pantharshit00** [reply]
👋 Hello @ggukggukee,  did you perhaps have a chance to check @austincrim's recommendation? Let us know if you still have any questions!
 
 Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
